generator/views/operational_monitoring_view.py

```diff
@@ -0,0 +1,102 @@
+"""Class to describe an Operational Monitoring View."""
+
+from textwrap import dedent
+from typing import Any, Dict, List, Optional, Set
+
+from .ping_view import PingView
+from .view import ViewDict
+
+# These are fields we don't need in our view
+EXCLUDED_FIELDS: Set[str] = {
+    "submission_date",
+    "client_id",
+    "build_id",
+    "branch",
+    "metrics",
+}
+
+
+class OperationalMonitoringView(PingView):
+    """A view on a operational monitoring table."""
+
+    type: str = "operational_monitoring_view"
+    dimensions: List[Dict[str, str]] = [
+        {
+            "name": "build_id",
+            "type": "date",
+            "sql": "PARSE_DATE('%Y%m%d', CAST(${TABLE}.build_id AS STRING))",
+        },
+        {"name": "branch", "type": "string", "sql": "${TABLE}.branch"},
+        {"name": "probe", "type": "string", "sql": "${TABLE}.name"},
+    ]
+
+    measures: List[Dict[str, str]] = [
+        {
+            "name": "median",
+            "type": "number",
+            "sql": dedent(
+                """
+            mozfun.hist.percentiles(
+              mozfun.hist.merge(
+                ARRAY_AGG(
+                  ${TABLE}.histograms IGNORE NULLS
+                )
+              ),
+              [0.5]
+            )[SAFE_OFFSET(0)].value
+            """
+            ),
+        }
+    ]
+
+    def __init__(self, namespace: str, name: str, tables: List[Dict[str, str]]):
+        """Create instance of a OperationalMonitoringView."""
+        super().__init__(namespace, name, tables)
+
+    @classmethod
+    def from_dict(klass, namespace: str, name: str, _dict: ViewDict) -> PingView:
+        """Get a OperationalMonitoringView from a dict representation."""
+        return OperationalMonitoringView(namespace, name, _dict["tables"])
+
+    def to_lookml(self, bq_client, v1_name: Optional[str]) -> Dict[str, Any]:
+        """Get this view as LookML."""
+        if len(self.tables) == 0:
+            raise Exception(
+                (f"Operational Monitoring view {self.name} " f"has no tables")
+            )
+
+        reference_table = self.tables[0]["table"]
+
+        schema = bq_client.get_table(reference_table).schema
+        for field in schema:
+            if field.name in EXCLUDED_FIELDS:
+                continue
+
+            self.dimensions.append(
+                {
+                    "name": field.name,
+                    "type": field.field_type,
+                    "sql": "${TABLE}." + f"{field.name}",
+                }
+            )
+
+        return {
+            "views": [
+                {
+                    "name": "fission    ",
+                    "derived_table": {
+                        "sql": dedent(
+                            f"""
+                            SELECT *
+                            FROM `{reference_table}`
+                            CROSS JOIN UNNEST(metrics)
```

Comment:
This nested field is going to make generation difficult. Can we expose this in the view, so that generation can work on just the raw table?

Reply:
Do you mind elaborating on what you mean here? Do you mean to expose it in the BQ view or in the looker view? What is difficult about this?

Reply:
So I actually don't want to include these other dimensions in the looker view since they wouldn't be needed. Perhaps I'm conflating what would be in the looker view vs. what would be in the explore. The explore wouldn't expose these extra dimensions so I didn't see a need to have them in the view either.